### PR TITLE
chore(chaos_result): Add ChaosPod label to ChaosResult

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: Udit Gaurav <udit.gaurav@mayadata.io>

### CODE REVIEW

**Comment by uditgaurav** [comment]
Choas Result:

```yaml
Name:         nginx-chaos-pod-delete
Namespace:    default
Labels:       chaosUID=14e625dd-0495-4fdd-8f48-63f398f76853
              controller-uid=2d283702-6c8a-4a83-95a6-bf80073341d8
              job-name=pod-delete-dy79hb
              name=nginx-chaos-pod-delete
Annotations:  <none>
API Version:  litmuschaos.io/v1alpha1
Kind:         ChaosResult
Metadata:
  Creation Timestamp:  2020-10-09T04:00:16Z
  Generation:          2
  Resource Version:    21170557
  Self Link:           /apis/litmuschaos.io/v1alpha1/namespaces/default/chaosresults/nginx-chaos-pod-delete
  UID:                 763a0a90-92aa-4300-b5d8-a7aa3ae072a5
Spec:
  Engine:      nginx-chaos
  Experiment:  pod-delete
Status:
  Experimentstatus:
    Fail Step:  N/A
    Phase:      Completed
    Verdict:    Pass
Events:
  Type    Reason   Age    From                     Message
  ----    ------   ----   ----                     -------
  Normal  Awaited  6m13s  pod-delete-dy79hb-5fmqw  experiment: pod-delete, Result: Awaited
  Normal  Pass     5m21s  pod-delete-dy79hb-5fmqw  experiment: pod-delete, Result: Pass

```

File: pkg/result/chaosresult.go [hunk 1]
```diff
@@ -40,7 +40,15 @@ func ChaosResult(chaosDetails *types.ChaosDetails, clients clients.ClientSets, r
...
 	if len(resultList.Items) == 0 {
-		return InitializeChaosResult(chaosDetails, clients, resultDetails)
+
+		// Getting chaos pod label and passing it in chaos result
+		chaosPod, err := clients.KubeClient.CoreV1().Pods(chaosDetails.ChaosNamespace).Get(chaosDetails.ChaosPodName, metav1.GetOptions{})
+		if err != nil {
+			return errors.Errorf("fail to get chaos pod err: %v", err)
+		}
+		chaosResultLabel := chaosPod.Labels
+		chaosResultLabel["name"] = resultDetails.Name
```
Comment (ksatchit) [inline pkg/result/chaosresult.go:50]: @uditgaurav the idea of this propagating the labels to chaosresult is to ensure that the chaos experiments resources (experiment cr def, experiment pod & result cr) share a common set of labels. In other words, we want to retain the 
`name: <experiment-name>` label and override with the result name.

**Review by ksatchit — DISMISSED** [review]

**Comment by ksatchit** [comment]
We might need some exp labels propagated to the helper pods too. cc: @ispeakc0de

File: chaoslib/litmus/container-kill/lib/container-kill.go [hunk 2]
```diff
@@ -117,7 +120,10 @@ func InjectChaosInParallelMode(experimentsDetails *experimentTypes.ExperimentDet
 	// creating the helper pod to perform container kill chaos
 	for _, pod := range targetPodList.Items {
 		runID := common.GetRunID()
-		err := CreateHelperPod(experimentsDetails, clients, pod.Name, pod.Spec.NodeName, runID)
+		// prepare label for helper pod the label is initialised while creating chaos result
+		chaosDetails.ExperimentLabel["name"] = experimentsDetails.ExperimentName + "-" + runID
```
Comment (ispeakc0de) [inline chaoslib/litmus/container-kill/lib/container-kill.go:124]: Why did we separate this? removed from helper pod and adding here?

Comment (uditgaurav) [reply chaoslib/litmus/container-kill/lib/container-kill.go:124]: label with key `name` is coming from chaos pod also we need to override it.

Comment (ispeakc0de) [reply chaoslib/litmus/container-kill/lib/container-kill.go:124]: same question if the helper pod have different labels then why are we reusing the `ExperimentLabel` variable instead we can follow the older approach and just add `app.kubernetes.io/part-of: litmus`

Comment (ksatchit) [reply chaoslib/litmus/container-kill/lib/container-kill.go:124]: Agree

File: pkg/result/chaosresult.go [hunk 3]
```diff
@@ -101,8 +108,16 @@ func InitializeChaosResult(chaosDetails *types.ChaosDetails, clients clients.Cli
...
 			"name": resultDetails.Name,
 		}
+		// Getting chaos pod label and passing it in chaos result
+		chaosPod, err := clients.KubeClient.CoreV1().Pods(chaosDetails.ChaosNamespace).Get(chaosDetails.ChaosPodName, v1.GetOptions{})
+		if err != nil {
+			return errors.Errorf("fail to get chaos pod err: %v", err)
+		}
+		experimentLabel := chaosPod.Labels
+		experimentLabel["name"] = resultDetails.Name
+
```
Comment (ispeakc0de) [inline pkg/result/chaosresult.go:118]: why are we deriving this again? can we reuse `chaosResultLabel` var?

File: pkg/result/chaosresult.go [hunk 4]
```diff
@@ -127,7 +142,7 @@ func GetProbeStatus(resultDetails *types.ResultDetails) []v1alpha1.ProbeStatus {
 }
 
 //PatchChaosResult Update the chaos result
-func PatchChaosResult(result *v1alpha1.ChaosResult, clients clients.ClientSets, chaosDetails *types.ChaosDetails, resultDetails *types.ResultDetails) error {
+func PatchChaosResult(result *v1alpha1.ChaosResult, clients clients.ClientSets, chaosDetails *types.ChaosDetails, resultDetails *types.ResultDetails, chaosResultLabel map[string]string) error {
```
Comment (ispeakc0de) [inline pkg/result/chaosresult.go:145]: update the labels inside this function as well

File: pkg/result/chaosresult.go [hunk 5]
```diff
@@ -102,7 +109,7 @@ func InitializeChaosResult(chaosDetails *types.ChaosDetails, clients clients.Cli
 			"name": resultDetails.Name,
 		}
```
Comment (ispeakc0de) [inline pkg/result/chaosresult.go:110]: Remove 107-110 section as well. It doesn't need initialization now

**Review by ispeakc0de — APPROVED** [review]

**Review by ksatchit — APPROVED** [review]

**Pull request merged**